dianna/methods/shap.py

```diff
@@ -1,21 +1,223 @@
 import numpy as np
+import shap
+import onnx
+from skimage.segmentation import slic
+from onnx_tf.backend import prepare  # onnx to tf model converter&runner
+import warnings
 
 
-class SHAP:
+class KernelSHAP:
     """
-    SHAP implementation based on captum https://github.com/pytorch/captum
+    Kernel SHAP implementation based on shap https://github.com/slundberg/shap
     """
-    def __init__(self, shap_arg_1=0, shap_arg_2=0):
+
+    def __init__(self):
+        """KernelSHAP initializer.
+
         """
-        Example constructor for testing.
+
+    def _segment_image(
+        self,
+        image,
+        n_segments,
+        compactness,
+        max_num_iter,
+        sigma,
+        spacing,
+        convert2lab,
+        enforce_connectivity,
+        min_size_factor,
+        max_size_factor,
+        slic_zero,
+        start_label,
+        mask,
+        channel_axis_first,
+    ):
+        """Create segmentation to explain by segment, not every pixel
+
+        This could help speed-up the calculation when the input size is very large.
+
+        This function segments image using k-means clustering in Color-(x,y,z) space. It uses
+        scikit-image.
+
+        Args:
+            Check keyword arguments for the skimage.segmentation.slic function
+            via the following link:
+            https://scikit-image.org/docs/dev/api/skimage.segmentation.html#skimage.segmentation.slic
         """
-        self.shap_arg_1 = shap_arg_1
-        self.shap_arg_2 = shap_arg_2
 
-    def explain_image(self, model, input_data):
+        if channel_axis_first:
+            image = np.transpose(image, (1, 2, 0))
+
+        image_segments = slic(
```

Comment:
```suggestion
        image_segments = skimage.segmentation.slic(
```
This makes it easier to read for people that don't know `skimage.segmentation` by heart.

Reply:
Sounds good! Corrected!